pinot-segment-local/src/main/java/org/apache/pinot/segment/local/realtime/converter/RealtimeSegmentConverter.java

```diff
@@ -94,14 +93,14 @@ public void build(@Nullable SegmentVersion segmentVersion, ServerMetrics serverM
     }
     if (_noDictionaryColumns != null) {
       genConfig.setRawIndexCreationColumns(_noDictionaryColumns);
-      Map<String, ChunkCompressionType> columnToCompressionType = new HashMap<>();
       for (String column : _noDictionaryColumns) {
         FieldSpec fieldSpec = _dataSchema.getFieldSpecFor(column);
+        // Compression type is already configured in the constructor of genConfig.
+        // Only update metrics compression type.
         if (fieldSpec.getFieldType().equals(FieldSpec.FieldType.METRIC)) {
-          columnToCompressionType.put(column, ChunkCompressionType.PASS_THROUGH);
+          genConfig.setRawIndexCompressionType(column, ChunkCompressionType.PASS_THROUGH);
         }
       }
```

Comment:
I think this part of the logic can be removed completely. In `SegmentColumnarIndexCreator.getColumnCompressionType()`, the compression type for metrics are default to `PASS_THROUGH`

Reply:
Thanks. You mean I can delete this whole block? (line 96-103)?

Reply:
Yes. We should not blindly override all metric compression to PASS_THROUGH even if it is explicitly configured differently